monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+                    transforms.LoadImaged(keys=["image", "label"]),
+                    transforms.EnsureChannelFirstd(keys=["image", "label"]),  # this creates label to be (1,H,W,D)
+                    transforms.Orientationd(keys=["image", "label"], axcodes="RAS"),
+                    transforms.EnsureTyped(keys=["image", "label"], data_type="tensor"),
+                    # some dataset has onehot label size of (H,W,D,C). Only allow label index of size (1,H,W,D)
+                    transforms.Lambdad(
+                        keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x
+                    ),
+                    transforms.SqueezeDimd(keys=["label"], dim=0),  # make label (H,W,D)
+                ]
+            ),
+        )
+        self.dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=worker, collate_fn=lambda x: x)
+        # If average all the modalities in the summary, if False, the stats for each modality are
+        # calculated separately
+        self.SUMMERY_AVERAGE = average
+        self.DO_CONNECTED_COMP = do_ccp
+        self.device = device
+        # gather all summary function for combining case stats
+        self.gather_summary: Dict[str, Dict] = {}
+        self._hardcode_functions()
+        self._hardcode_operations()
+        print(self.gather_summary)
+
+    def _hardcode_functions(self):
+        """Register functions for calculating stats."""
+        self.functions = [self._get_case_image_stats, self._get_case_foreground_image_stats, self._get_label_stats]
+        self.functions_summary = [
+            self._get_case_image_stats_summary,
+            self._get_case_foreground_image_stats_summary,
+            self._get_label_stats_summary,
+        ]
+
+    def _hardcode_operations(self):
+        """Register operations for basic data operation."""
+        # define basic operations for stats
+        self.operations = {
+            "max": torch.max,
+            "mean": torch.mean,
+            "median": torch.median,
+            "min": torch.min,
+            "percentile": partial(torch.quantile, q=[0.005, 0.10, 0.90, 0.995]),
+            # 'percentile_00_5' : partial(torch.quantile, q=.005),
+            # 'percentile_99_5' : partial(torch.quantile, q=.995),
+            # 'percentile_10_0' : partial(torch.quantile, q=.10),
+            # 'percentile_90_0' : partial(torch.quantile, q=.90),
+            "stdev": partial(torch.std, unbiased=False),
+        }
+        # allow mapping the output of self.operations to new keys (save computation time)
+        # the output from torch.quantile is mapped to four keys.
+        self.operations_mappingkey = {
+            "percentile": ["percentile_00_5", "percentile_10_0", "percentile_90_0", "percentile_99_5"]
+        }
+        # np version of self.operations. torch operation has inconsistent interfaces.
+        # only used in the summary part.
+        self.operations_np = {
```

Comment:
I don't quite understand why we need both `torch` operations and `numpy` operations for the data analysis, is it possible to simplify to only the `torch` as we recommend `MetaTensor` now?
@wyli @rijobro What do you think?

Thanks.

Reply:
According to comments in the original repository, torch operation may fail in some cases (torch.quantile was named as an example) when the input is too large.

Reply:
Hi @dongyang0122 @mingxin-zheng ,

Could you please help give a reproducible error example with PyTorch 1.12?
If it has some error,  I think we can report it to the PyTorch team.

Thanks in advance.

Reply:
for torch.quantile there's a util in monai https://github.com/Project-MONAI/MONAI/blob/a9095d09d919e94c2c07aa74fc46c1e4f9530ecd/monai/transforms/utils_pytorch_numpy_unification.py#L80

Reply:
The issue for torch.quantile https://github.com/pytorch/pytorch/issues/64947

Reply:
Hi @wyli ,

So what do you think about the `operations` here? Should we keep both numpy and torch APIs?

Thanks.

Reply:
agree that we should unify numpy and torch APIs `self.operations_np` and `self.operations`. perhaps for `"percentile"` we implement it with the numpy API and convert the output to tensors.

Reply:
Thanks @wyli . I incorporated your fix into mine and added a few more functions (min/max/...) to harmonize the numpy and torch calls. Operations_np is removed now. More in 2dc9e2e3df0a65bf0345f2be16da5a764183f7d2